editor/store/effects.js

```diff
@@ -254,6 +253,8 @@ export default {
 		dispatch( removeNotice( TRASH_POST_NOTICE_ID ) );
 		apiRequest( { path: `/wp/v2/${ basePath }/${ postId }`, method: 'DELETE' } ).then(
 			() => {
+				const post = getCurrentPost( getState() );
+				dispatch( resetPost( { ...post, status: 'trashed' } ) );
```

Comment:
This seems like what `updatePost` is meant to handle (updating subsets of post properties). Unfortunately we have it tangled with change detection so that it considers dirtiness as reset. This _might_ not be a problem if we expect a redirect to occur anyways (in fact maybe a good thing to avoid any possibility of a prompt).

Related: https://github.com/WordPress/gutenberg/pull/7130#issuecomment-394838613

Reply:
>This seems like what `updatePost` is meant to handle [...]

What do you think about this? Should we try to use `updatePost`? If it's at least the desirable end-solution, should we add a "TODO" comment or task to follow-up on this?

Reply:
I think it should be a todo for now because at some point I want to move the `currentPost` out of the `editor's` state into `core-data`, in which case we'd have a `updateEntityRecord` action